File: wheystore/wheystore.Web/Modules/SanPhamDB/TinhThanh/TinhThanhRow.cs
namespace wheystore.SanPhamDB;

[ConnectionKey("Default"), Module("SanPhamDB"), TableName("TinhThanh")]
[DisplayName("Tinh Thanh"), InstanceName("Tinh Thanh")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class TinhThanhRow : Row<TinhThanhRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Column("ID"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Ten Tinh Thanh"), Size(200), NotNull, QuickSearch, NameProperty]
    public string TenTinhThanh { get => fields.TenTinhThanh[this]; set => fields.TenTinhThanh[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField TenTinhThanh;

    }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/TinhThanhRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface TinhThanhRow {
    Id?: number;
    TenTinhThanh?: string;
}

export abstract class TinhThanhRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'TenTinhThanh';
    static readonly localTextPrefix = 'SanPhamDB.TinhThanh';
    static readonly lookupKey = 'SanPhamDB.TinhThanh';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TinhThanhRow>('SanPhamDB.TinhThanh') }
    static async getLookupAsync() { return getLookupAsync<TinhThanhRow>('SanPhamDB.TinhThanh') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<TinhThanhRow>();
}